velox/dwio/parquet/reader/PageReader.cpp

```diff
@@ -364,6 +365,32 @@ void PageReader::prepareDictionary(const PageHeader& pageHeader) {
           // We start from the end to allow in-place expansion.
           values[i] = parquetValues[i];
         }
+      } else if (
+          parquetType == thrift::Type::INT64 &&
+          type_->logicalType_.has_value()) {
+        auto logicalType = type_->logicalType_.value();
+        if (logicalType.__isset.TIMESTAMP) {
+          VELOX_CHECK(
+              logicalType.TIMESTAMP.isAdjustedToUTC,
+              "Only UTC adjusted Timestamp is supported");
+          auto values = dictionary_.values->asMutable<Timestamp>();
+          auto parquetValues = dictionary_.values->asMutable<char>();
+          int64_t units;
+
+          if (logicalType.TIMESTAMP.unit.__isset.MICROS) {
+            for (auto i = dictionary_.numValues - 1; i >= 0; --i) {
+              memcpy(&units, parquetValues + i * typeSize, typeSize);
```

Comment:
Before memcpy, do we need to allocate memory for `dictionary_.values` referring to Velox type length instead of typeSize (just like the case when `type_->type()->isShortDecimal() && parquetType == thrift::Type::INT32`)? The size of Timestamp should be larger than the size of int64_t.

Reply:
Good point. Timestamp size is double of int64_t. Wonder why it didn't gave error in tests.